pkg/daemon/daemon.go

```diff
@@ -93,398 +55,304 @@ func New(
+	return reqReboot, reqDrain, nil
+}
 
-		err = systemd.WriteSriovSupportedNics()
-		if err != nil {
-			log.Log.Error(err, "nodeStateSyncHandler(): failed to write supported nic ids file for systemd mode")
-			return err
-		}
+func (dn *DaemonReconcile) checkSystemdStatus(ctx context.Context, desiredNodeState *sriovnetworkv1.SriovNetworkNodeState) (*systemd.SriovResult, bool, error) {
+	if !vars.UsingSystemdMode {
+		return nil, false, nil
 	}
 
-	log.Log.V(0).Info("nodeStateSyncHandler(): aggregated daemon",
-		"drain-required", reqDrain, "reboot-required", reqReboot, "disable-drain", dn.disableDrain)
+	funcLog := log.Log.WithName("checkSystemdStatus")
+	serviceEnabled, err := dn.HostHelpers.IsServiceEnabled(systemd.SriovServicePath)
+	if err != nil {
+		funcLog.Error(err, "failed to check if sriov-config service exist on host")
+		return nil, false, err
+	}
+	postNetworkServiceEnabled, err := dn.HostHelpers.IsServiceEnabled(systemd.SriovPostNetworkServicePath)
+	if err != nil {
+		funcLog.Error(err, "failed to check if sriov-config-post-network service exist on host")
+		return nil, false, err
+	}
 
-	// handle drain only if the plugin request drain, or we are already in a draining request state
-	if reqDrain || !utils.ObjectHasAnnotation(dn.desiredNodeState,
-		consts.NodeStateDrainAnnotationCurrent,
-		consts.DrainIdle) {
-		drainInProcess, err := dn.handleDrain(reqReboot)
+	// if the service doesn't exist we should continue to let the k8s plugin to create the service files
+	// this is only for k8s base environments, for openshift the sriov-operator creates a machine config to will apply
+	// the system service and reboot the node the config-daemon doesn't need to do anything.
+	sriovResult := &systemd.SriovResult{SyncStatus: consts.SyncStatusFailed,
+		LastSyncError: fmt.Sprintf("some sriov systemd services are not available on node: "+
+			"sriov-config available:%t, sriov-config-post-network available:%t", serviceEnabled, postNetworkServiceEnabled)}
+	exist := false
+
+	// check if the service exist
+	if serviceEnabled && postNetworkServiceEnabled {
+		sriovResult, exist, err = systemd.ReadSriovResult()
 		if err != nil {
-			log.Log.Error(err, "failed to handle drain")
-			return err
-		}
-		if drainInProcess {
-			return nil
+			funcLog.Error(err, "failed to load sriov result file from host")
+			return nil, false, err
 		}
 	}
 
+	//// only if something is not equal we apply if not we continue to check if something change on the node,
+	//// and we need to trigger a reconfiguration
+	//if desiredNodeState.Status.SyncStatus != sriovResult.SyncStatus ||
+	//	desiredNodeState.Status.LastSyncError != sriovResult.LastSyncError {
+	//	err = dn.updateSyncState(ctx, desiredNodeState, sriovResult.SyncStatus, sriovResult.LastSyncError)
+	//	if err != nil {
+	//		funcLog.Error(err, "failed to update sync status")
+	//	}
+	//	return sriovResult, err
+	//}
+
+	// TODO: check if we need this
```

Comment:
i believe we need.

in case systemd services were enabled, and there is error in sriovResult, we want to reflect that in status and stop reconcile.

Reply:
@ykulazhenkov for some more feedback on this bit

Reply:
it was moved out of this function so we should be ok I remove this